logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationListDetailsUseCase.kt

```diff
@@ -10,17 +11,35 @@ import kotlinx.coroutines.flow.map
 
 class ObserveConversationListDetailsUseCase(
     private val conversationRepository: ConversationRepository,
-    private val syncManager: SyncManager
+    private val syncManager: SyncManager,
+    private val callRepository: CallRepository
 ) {
 
     suspend operator fun invoke(): Flow<List<ConversationDetails>> {
         syncManager.startSyncIfIdle()
-        return conversationRepository.observeConversationList().map { conversations ->
+
+        val conversationsFlow = conversationRepository.observeConversationList().map { conversations ->
             conversations.map { conversation ->
                 conversationRepository.observeConversationDetailsById(conversation.id)
             }
         }.flatMapLatest { flowsOfDetails ->
             combine(flowsOfDetails) { latestValues -> latestValues.asList() }
         }
+
+        return combine(
+            conversationsFlow,
+            callRepository.ongoingCallsFlow()
+        ) { conversations, calls ->
+            conversations.map {
+                when (it) {
+                    is ConversationDetails.Self,
+                    is ConversationDetails.Connection,
+                    is ConversationDetails.OneOne-> it
+                    is ConversationDetails.Group -> it.copy(
+                        hasOngoingCall = (it.conversation.id in calls.map { call -> call.conversationId })
+                    )
+                }
+            }
+        }
```

Comment:
Could you write tests for this?

##### Given an ongoing call, when fetching conversation details, then the conversation should have ongoing call

1. Return a conversation from the mocked ConversationRepo
2. Return an ongoing call on this conversation from the mocked CallRepo
3. Assert it `hasOngoingCall == true`

##### Given a conversation without ongoing call, when fetching conversation details, then the conversation should not have ongoing call

1. Return a conversation from the mocked ConversationRepo
2. Return no ongoing calls from the mocked CallRepo
3. Assert it `hasOngoingCall != true`